orangecontrib/prototypes/widgets/owcontingency.py

```diff
@@ -15,32 +23,45 @@ class OWContingencyTable(widget.OWWidget):
     priority = 2010
 
     inputs = [("Data", Table, "set_data", widget.Default)]
-    outputs = [("Contingency Table", Table, )]
+    outputs = [("Contingency Table", Table, ),
+               ("Selected Data", Table),
+               (ANNOTATED_DATA_SIGNAL_NAME, Table)]
 
     settingsHandler = DomainContextHandler(metas_in_res=True)
     rows = ContextSetting(None)
     columns = ContextSetting(None)
+    selection = ContextSetting(set())
     auto_apply = Setting(True)
 
-    want_main_area = False
+    want_main_area = True
 
     def __init__(self):
         super().__init__()
 
         self.data = None
         self.feature_model = DomainModel(valid_types=DiscreteVariable)
+        self.table = None
 
         box = gui.vBox(self.controlArea, "Rows")
         gui.comboBox(box, self, 'rows', sendSelectedValue=True,
-                     model=self.feature_model, callback=self._invalidate)
+                     model=self.feature_model, callback=self._attribute_changed)
 
         box = gui.vBox(self.controlArea, "Columns")
         gui.comboBox(box, self, 'columns', sendSelectedValue=True,
-                     model=self.feature_model, callback=self._invalidate)
+                     model=self.feature_model, callback=self._attribute_changed)
+
+        gui.rubber(self.controlArea)
+
+        box = gui.vBox(self.controlArea, "Scores")
+        self.scores = gui.widgetLabel(box, "\n\n")
 
         self.apply_button = gui.auto_commit(
             self.controlArea, self, "auto_apply", "&Apply", box=False)
 
+        self.tablemodel = QStandardItemModel(self)
+        view = self.tableview = ContingencyTable(self, self.tablemodel)
```

Comment:
Not sure why you initialize QStandardItemModel here. You could simplify this by moving it to ContingencyTable class.